File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/scheduling/SchedulingServiceFactory.java
package com.neatorobotics.sdk.android.robotservices.scheduling;

/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public class SchedulingServiceFactory {
    public static SchedulingService get(String serviceVersion) {
        if("basic-1".equalsIgnoreCase(serviceVersion)) return new SchedulingBasic1Service();
        else if("basic-2".equalsIgnoreCase(serviceVersion)) return new SchedulingBasic2Service();
        else if("minimal-1".equalsIgnoreCase(serviceVersion)) return new SchedulingMinimal1Service();
        else return null;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/scheduling/SchedulingServiceFactory.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.scheduling

import com.neatorobotics.sdk.android.robotservices.RobotServices

object SchedulingServiceFactory {
    fun get(serviceVersion: String): SchedulingService? {
        return when {
            RobotServices.VERSION_BASIC_1.equals(serviceVersion, ignoreCase = true) -> SchedulingBasic1Service()
            RobotServices.VERSION_MINIMAL_1.equals(serviceVersion, ignoreCase = true) -> SchedulingMinimal1Service()
            RobotServices.VERSION_BASIC_2.equals(serviceVersion, ignoreCase = true) -> SchedulingBasic2Service()
            RobotServices.VERSION_ADVANCED_2.equals(serviceVersion, ignoreCase = true) -> SchedulingAdvanced2Service()
            else -> null
        }
    }
}
